conjure-java-jackson-serialization/src/main/java/com/palantir/conjure/java/serialization/ObjectMappers.java

```diff
@@ -205,7 +205,7 @@ public static <M extends ObjectMapper, B extends MapperBuilder<M, B>> B withDefa
      * </ul>
      */
     public static ObjectMapper withDefaultModules(ObjectMapper mapper) {
-        return mapper.setTypeFactory(new NonCachingTypeFactory())
+        return mapper.setTypeFactory(NonCachingTypeFactory.from(mapper.getTypeFactory()))
```

Comment:
I considered unsetting `IGNORE_DUPLICATE_MODULE_REGISTRATIONS` and always using `NonCachingTypeFactory`, however that's unsound if modules that CJR doesn't apply which add TypeModifier instances have already been registered.

The most common case is that the `new*Mapper` methods are used initially, so I think what I've done here is the most reasonable trade-off.